src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const onlyMen = people.filter(item =>
+    item.sex === 'm'
+  );
+
+  const ageInEighteen = onlyMen.filter(item =>
+    Math.ceil(item.died / 100) === century)
+    .map(item => item.died - item.born);
+
+  return ageInEighteen.reduce((sum, age) =>
+    sum + age / ageInEighteen.length, 0)
+    || onlyMen.map(item => item.died - item.born)
+      .reduce((sum, age) =>
+        sum + age / onlyMen.length, 0);
```

Comment:
It hard to read
```suggestion
  return (
    ageInEighteen.reduce((sum, age) => sum + age / ageInEighteen.length, 0) ||
    onlyMen
      .map((item) => item.died - item.born)
      .reduce((sum, age) => sum + age / onlyMen.length, 0)
  );
```
or better write it to a variable and return this variable